### PR TITLE
React test

### PR DESCRIPTION
**Pull request by lmd59**
Pull requests into Cypress require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [ ] This pull request describes why these changes were made.
- [ ] Internal ticket for this PR:
- [ ] Internal ticket links to this PR
- [ ] Code diff has been done and been reviewed
- [ ] Tests are included and test edge cases
- [ ] Tests have been run locally and pass

**Reviewer 1:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

**Comment by dczulada** [comment]
Looks really good.  Looks like what I did in the Error report has made the error displays a little wonky.

File: app/views/records/show.html.erb [hunk 1]
```diff
@@ -83,6 +83,7 @@
       <% end %>
       </ul>
     </div>
-    
+    <% patient_resource = @record.patient.entry.find { |e| e.resource.resourceType == 'Patient' }.resource %>
+    <%= react_component("PatientView", { data: patient_resource, patient: patient_resource, entry: @relevant_entries, errors: @record.entry_id_validation_hash }) %>
```
Comment (dczulada) [inline app/views/records/show.html.erb]: Looks really good.  Looks like what I did in the Error report has made the error displays a little wonky.

Really Gross Screenshot
Screen Shot 2021-04-21 at 3 20 33 PM

We can probably strip out the 'entry' and 'resource' elements from the entry_id_validation_hash.  This code works, but you might have a better idea.

`<%= react_component("PatientView", { data: patient_resource, patient: patient_resource, entry: @relevant_entries, errors: @record.entry_id_validation_hash.map { |vh| [vh[0], vh[1].slice('errors', 'warnings', 'information')] }.to_h }) %>`

This makes the error pop-up look like this.

Screen Shot 2021-04-21 at 3 20 57 PM

**Review by dczulada — CHANGES_REQUESTED** [review]

File: app/javascript/packs/hello_react.jsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
+// Run this example by adding <%= javascript_pack_tag 'hello_react' %> to the head of your layout file,
+// like app/views/layouts/application.html.erb. All it does is render <div>Hello React</div> at the bottom
+// of the page.
+
+import React from 'react'
+import ReactDOM from 'react-dom'
+import PropTypes from 'prop-types'
+
+const Hello = props => (
+  <div>Hello {props.name}!</div>
```
Comment (dczulada) [inline app/javascript/packs/hello_react.jsx]: Looks like this file can probably be removed

File: app/javascript/packs/application.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+/* eslint no-console:0 */
+// This file is automatically compiled by Webpack, along with any other files
+// present in this directory. You're encouraged to place your actual application logic in
+// a relevant structure within app/javascript and only use these pack files to reference
+// that code so it'll be compiled.
+//
+// To reference this file, add <%= javascript_pack_tag 'application' %> to the appropriate
+// layout file, like app/views/layouts/application.html.erb
+
+console.log('Hello World from Webpacker')
```
Comment (dczulada) [inline app/javascript/packs/application.js]: looks like this line can be removed
